omod/src/main/webapp/resources/scripts/conditionlist/controllers/editcondition.controller.js

```diff
@@ -92,7 +95,12 @@ function ConditionController($scope, RestfulService, ConditionModel, ConceptMode
         var groups = document.getElementsByClassName("group");
         if($scope.condition.status == INACTIVE_STATUS){
             groups[2].style.visibility = "visible";
-            inputs[3].value = CommonFunctions.formatDate($scope.condition.endDate); 
+
+            if($scope.condition.endDate==null){
+                inputs[3].value = CommonFunctions.formatDate(new Date()); 
```

Comment:
Instead of getting your input element as `document.getElementsByTagName('input')[3]`, please use a lookup mechanism that is less brittle. I would recommend looking the element up by ID (giving it an ID if it doesn't have one).

To illustrate the problem: imagine that I, as a customization to my OpenMRS implementation, override the title bar to have it include an inline patient search box. Then each element of `document.getElementsByTagName('input')` that you're looking for has had its index offset by one, because the first input is now my search box. So your endDate input box will now be `inputs[4]`, and the conditions page won't work at all. Which will be very confusing to me, since I didn't mess with the conditions widget at all.

Reply:
Agree with @brandones ... I receive that this is outside the scope of this PR, but we really need to fix this to use a less-brittle method, mainly by adding an id attribute to the input elements as Brandon suggests.